Sources/AWSLambdaEvents/Utils/DateWrappers.swift

```diff
@@ -84,34 +113,256 @@ public struct RFC5322DateTimeCoding: Decodable, Sendable {
 
     public init(from decoder: Decoder) throws {
         let container = try decoder.singleValueContainer()
-        var string = try container.decode(String.self)
-        // RFC5322 dates sometimes have the alphabetic version of the timezone in brackets after the numeric version. The date formatter
-        // fails to parse this so we need to remove this before parsing.
-        if let bracket = string.firstIndex(of: "(") {
-            string = String(string[string.startIndex..<bracket].trimmingCharacters(in: .whitespaces))
+        let string = try container.decode(String.self)
+
+        do {
+            if #available(macOS 12.0, *) {
+                self.wrappedValue = try Date(string, strategy: RFC5322DateParseStrategy())
+            } else {
+                self.wrappedValue = try RFC5322DateParseStrategy().parse(string)
+            }
+        } catch {
+            throw DecodingError.dataCorruptedError(
+                in: container,
+                debugDescription:
+                    "Expected date to be in RFC5322 date-time format, but `\(string)` is not in the correct format"
+            )
         }
-        for formatter in Self.dateFormatters {
-            if let date = formatter.date(from: string) {
-                self.wrappedValue = date
-                return
+    }
+}
+
+struct RFC5322DateParsingError: Error {}
+
+struct RFC5322DateParseStrategy {
```

Comment:
Do we have any tests for this? This is a lot of untested code, and a lot of unsafe code that we should really ensure works properly if there are not other alternatives

Reply:
Yeah that's fair, it is currently being tested indirectly through the existing tests that parse dates.

Reply:
We can add more test for some of edge cases. I am also not sure if we need to support more than the handful of hard-coded named timezones I added. My assumption is though that S3 will always retuned well-formed dates in UTC.

Reply:
A regarding "unsafe". I think all API is actually safe. While we are operating on an "Unsafe"BufferPointer, we are only using its iterator methods, which should guarantee that there is no way to escape the buffer bounds by doing so.

Reply:
I did suggest to the Foundation team a while back to include an HTTP datetime FormatStyle in FoundationEssentials and they seemed open to the idea. I never chased it up though. @t089 would you be open to pitching it with a version of this code?